backend/app/infrastructure/external_api/safe/message_details.py

```diff
@@ -0,0 +1,27 @@
+import requests
+
+import app as app_module
+from app.constants import SAFE_API_MAINNET, SAFE_API_SEPOLIA
+from app.exceptions import ExternalApiException
+
+
+def get_message_details(message_hash: str, is_mainnet: bool) -> dict:
+    api_url = _get_api_url(message_hash, is_mainnet)
+
+    try:
+        response = requests.request("GET", api_url)
+        response.raise_for_status()
+        json_response = response.json()
+    except requests.exceptions.RequestException as e:
+        app_module.ExceptionHandler.print_stacktrace(e)
+        raise ExternalApiException(api_url, e, 500)
+
+    return json_response
+
+
+def _get_api_url(
+    message_hash: str,
+    is_mainnet: bool,
+) -> str:
+    base_url = SAFE_API_MAINNET if is_mainnet else SAFE_API_SEPOLIA
```

Comment:
we could build this url based on the current network name dynamically, so it'd work after we move to other testnet/L2 too. you can change it if it's a small tweak, otherwise let's leave it as it is for now

Reply:
We already use this flow in other places in code (checking chain_id and choosing a proper strategy for specific network). Changing the way of doing it is a great idea and I'm for it but it'd be the best to unify whole codebase, not only my changes - just for the sake of consistency and being neat. I wrote down this need and will be keeping it in the back of my head when adjusting to other networks. Makes senes to you?